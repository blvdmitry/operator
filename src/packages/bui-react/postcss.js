// Re-export of the config provided by bui-postcss package
const config = require("@bookingcom/bui-postcss");

module.exports = config.default;
