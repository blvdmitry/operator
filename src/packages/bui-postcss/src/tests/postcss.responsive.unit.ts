/* eslint-disable import/no-extraneous-dependencies */
import postcss from "postcss";

const responsivePlugin = require("../postcss.responsive");

test("postcss.apply", async () => {
  const input = `
    @custom-media --bui_small_viewport (max-width: 575px);
    @custom-media --bui_medium_viewport (min-width: 576px);
    @custom-media --bui_large_viewport (min-width: 1024px);
    @custom-media --bui_xlarge_viewport (min-width: 1280px);

    @bui-responsive .container--variant {
      @bui-modifier primary, secondary {
        color: red;
      }
      
      @bui-modifier primary {
        padding: var(--bui_spacing_4x);
        background: red;
      }

      @bui-modifier secondary {
        padding: var(--bui_spacing_6x);
      }
    }
    
    @bui-responsive .container--elevated {
      @bui-modifier true {
        border: var(--bui_border_width_200) solid var(--bui_color_accent_border);
      }

      @bui-modifier false {
        border: none;
      }
    }
    
    @bui-responsive .container--direction > *:nth-child(n) {
      @bui-modifier column {
        margin-block-start: auto;
        margin-inline-start: initial;
      }
    }
    
    @bui-responsive .container--direction > .item--split {
      @bui-modifier column {
        margin-block-start: auto;
        margin-inline-start: initial;
      }
    }
    
    @bui-responsive * > .container--direction {
      @bui-modifier column {
        margin-block-start: auto;
        margin-inline-start: initial;
      }
    }
  `;
  const output = await postcss([responsivePlugin()]).process(input, {
    from: "__placeholder-name.css",
  });

  expect(output.css).toMatchSnapshot();
});
