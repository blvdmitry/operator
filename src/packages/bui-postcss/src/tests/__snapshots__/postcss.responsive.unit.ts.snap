// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss.apply 1`] = `
"
    @custom-media --bui_small_viewport (max-width: 575px);
    @custom-media --bui_medium_viewport (min-width: 576px);
    @custom-media --bui_large_viewport (min-width: 1024px);
    @custom-media --bui_xlarge_viewport (min-width: 1280px);

    .container--variant-primary,.container--variant-secondary{
        color: red
}

    .container--variant-primary{
        color: initial;
        padding: var(--bui_spacing_4x);
        background: red
}

    .container--variant-secondary{
        color: initial;
        background: initial;
        padding: var(--bui_spacing_6x)
}

    @media (--bui_medium_viewport){

        .container--variant-primary--m,.container--variant-secondary--m{
        color: red
        }

        .container--variant-primary--m{
        color: initial;
        padding: var(--bui_spacing_4x);
        background: red
        }

        .container--variant-secondary--m{
        color: initial;
        background: initial;
        padding: var(--bui_spacing_6x)
        }
}

    @media (--bui_large_viewport){

        .container--variant-primary--l,.container--variant-secondary--l{
        color: red
        }

        .container--variant-primary--l{
        color: initial;
        padding: var(--bui_spacing_4x);
        background: red
        }

        .container--variant-secondary--l{
        color: initial;
        background: initial;
        padding: var(--bui_spacing_6x)
        }
}

    @media (--bui_xlarge_viewport){

        .container--variant-primary--xl,.container--variant-secondary--xl{
        color: red
        }

        .container--variant-primary--xl{
        color: initial;
        padding: var(--bui_spacing_4x);
        background: red
        }

        .container--variant-secondary--xl{
        color: initial;
        background: initial;
        padding: var(--bui_spacing_6x)
        }
}
    
    .container--elevated-true{
        border: var(--bui_border_width_200) solid var(--bui_color_accent_border)
}
    
    .container--elevated-false{
        border: none
}
    
    @media (--bui_medium_viewport){

        .container--elevated-true--m{
        border: var(--bui_border_width_200) solid var(--bui_color_accent_border)
        }

        .container--elevated-false--m{
        border: none
        }
}
    
    @media (--bui_large_viewport){

        .container--elevated-true--l{
        border: var(--bui_border_width_200) solid var(--bui_color_accent_border)
        }

        .container--elevated-false--l{
        border: none
        }
}
    
    @media (--bui_xlarge_viewport){

        .container--elevated-true--xl{
        border: var(--bui_border_width_200) solid var(--bui_color_accent_border)
        }

        .container--elevated-false--xl{
        border: none
        }
}
    
    .container--direction-column > *:nth-child(n){
        margin-block-start: auto;
        margin-inline-start: initial
}
    
    @media (--bui_medium_viewport){

        .container--direction-column--m > *:nth-child(n){
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_large_viewport){

        .container--direction-column--l > *:nth-child(n){
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_xlarge_viewport){

        .container--direction-column--xl > *:nth-child(n){
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    .container--direction-column > .item--split{
        margin-block-start: auto;
        margin-inline-start: initial
}
    
    @media (--bui_medium_viewport){

        .container--direction-column--m > .item--split{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_large_viewport){

        .container--direction-column--l > .item--split{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_xlarge_viewport){

        .container--direction-column--xl > .item--split{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    * > .container--direction-column{
        margin-block-start: auto;
        margin-inline-start: initial
}
    
    @media (--bui_medium_viewport){

        * > .container--direction-column--m{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_large_viewport){

        * > .container--direction-column--l{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
    
    @media (--bui_xlarge_viewport){

        * > .container--direction-column--xl{
        margin-block-start: auto;
        margin-inline-start: initial
        }
}
  "
`;
